spark/src/main/java/io/ddf/spark/SparkDDFManager.java

```diff
@@ -272,8 +272,23 @@ public DDF copyFrom(DDF ddf) throws DDFException {
       if (ddf instanceof S3DDF) {
         S3DDF s3DDF = (S3DDF) ddf;
         S3DataSourceCredentials cred = s3DDF.getManager().getCredential();
-        uri = String.format("s3a://%s:%s@%s/%s", cred.getAwsKeyID(), cred.getAwsScretKey(),
-            s3DDF.getBucket(), s3DDF.getKey());
+        if (s3DDF.getPath() != null) {
+          String[] fileList;
+          String path = s3DDF.getPath();
+          fileList = path.split(",");
```

Comment:
That pattern is already supported by Spark/Hadoop. The issue is that since we additionally separate the protocol and the FS path parts of the uri in our APIs, we can only easily support either of:
- `full path, full path`: by keeping splitting by comma.
- `path/{subpath1, subpath2}/`: by not doing any additional processing on the path.

If we support passing a list of paths, the second one would be more useful. If we don't, the first one is more useful.

To support list of paths:
- PyClient/RClient accepts either a string or a list of string
- PE HTTP APIs accept either `path` or `paths` parameters
- DDF core supports `paths` when constructing DDFs, deprecating `path`